v10.10 Changelog (#8063)
mattermost/docs: source/about/mattermost-v10-changelog.md

```
(release-v10.10-feature-release)=
## Release v10.10 - [Feature Release](https://docs.mattermost.com/about/release-policy.html#release-types)

**Release Day: July 16, 2025**

### Important Upgrade Notes
 - Added a new column ``DefaultCategoryName`` to the ``Channels`` table. This is nullable and stores a category name to be added/created when new users join a channel. This is only used if the ``ExperimentalChannelCategorySetting`` is enabled. No database downtime is expected for this upgrade. See the [Important Upgrade Notes](https://docs.mattermost.com/upgrade/important-upgrade-notes.html) for more details.

```{Important}
If you upgrade from a release earlier than v10.9, please read the other [Important Upgrade Notes](https://docs.mattermost.com/upgrade/important-upgrade-notes.html).
```

### Improvements

#### User Interface (UI)
 - Pre-packaged Calls plugin [v1.9.1](https://github.com/mattermost/mattermost-plugin-calls/releases/tag/v1.9.1).
 - Pre-packaged GitLab plugin [v1.10.0](https://github.com/mattermost/mattermost-plugin-gitlab/releases/tag/v1.10.0).
 - Pre-packaged GitHub plugin [v2.4.0](https://github.com/mattermost/mattermost-plugin-github/releases/tag/v2.4.0).

>  - Pre-packaged Boards plugin [v9.1.3](https://github.com/mattermost/mattermost-plugin-boards/releases/tag/v9.1.3).

Reviewer: This should be v9.1.4 once the cherry pick is complete https://github.com/mattermost/mattermost/pull/31771 

cc// @Rajat-Dabade 

```suggestion
 - Pre-packaged Boards plugin [v9.1.4](https://github.com/mattermost/mattermost-plugin-boards/releases/tag/v9.1.4).
```

 - Resolved inconsistent styling issues in new integration pages.
 - Improved the visual styling of blockquotes and comment threads for better readability and a modern appearance.
 - Improved screen reader support for autocompletes and channel switcher.
 - Added an aria-live region for improved accessibility of file preview modal carousel position updates.
 - Add focusability to react-select remove button in notifications settings.
 - Updated **Profile** settings client-side validation to use more modern/accessible paradigms.
 - Enhanced the accessibility of login and password reset functionality.
 - Stopped the Threads textbox from automatically taking focus when it contained a draft.
 - Added a user setting to toggle rendering of emoticons (e.g., :D) as emojis (e.g., 😄).
 - Added support for the ``from:`` search filter in cross-team searches.
 - Standardized button styling by consolidating color variables and removing redundant theme definitions.
 - Ignored email domain in user searches on the client side.
 - Automatically detected and updated timezone changes without requiring a client refresh.

#### Administration
 - Added support for group messages in Shared Channels.

>  - Added support for licenses that enforce seat counts with 5% grace period.
>  - Replaced fixed 5% grace period for user seat limits with configurable ``ExtraUsers`` field for exact control over additional seats.

Reviewer 2: ```suggestion
 - Added support for licenses that enforce seat counts with a configurable ``ExtraUsers`` field for exact control over allowed overages.
```

#### Changes to all plans:
 - Under ``ExperimentalSettings`` in ``config.json``:
    - Added ``ExperimentalChannelCategorySorting`` configuration setting to add the ability to automatically sort channels into categories upon creation/renaming.

#### Changes to Enterprise plan:
 - Under ``DataRetentionSettings`` in ``config.json``:
    - Added ``PreservePinnedPosts`` configuration setting. If it's set to ``true``, pinned posts will not be deleted by data retention. 

#### Changes to Enterprise Advanced plan:
 - Under ``ConnectedWorkspacesSettings`` in ``config.json``:
    - Added ``MemberSyncBatchSize``, ``SyncUsersOnConnectionOpen``, ``GlobalUserSyncBatchSize`` configuration settings to allow remote users to be discoverable in the Direct/Group Message modal. 

### API Changes
 - Added property fields and value methods to the plugin API.
 - Improved the semantics of Groups API errors when invalid parameters were specified.

### Open Source Components
 - Added ``mholt/archives`` and removed ``code.sajari.com/docconv`` from https://github.com/mattermost/mattermost.

### Go Version
 - v10.10 is built with Go ``v1.24.3``.

> 
> ### Known Issues
>  - Setting the license file location through an environment variable still gives the option to upload a new license through the System Console, resulting in the license being overwritten by the one set through the environment variable. See this [knowledge base article](https://support.mattermost.com/hc/en-us/articles/33911983851284-System-console-still-displays-old-license-after-uploading-a-new-one) on how to resolve this issue.
> 

Reviewer 2: I think I asked about this last time, but "why this issue?" -- surely there are hundreds of known issues that we've left unresolved, and I honestly don't think anyone is even working on this one. Should we just remove `Known Issues`?
```suggestion
```
